matrix/matrix_binary_search.py

```diff
@@ -0,0 +1,52 @@
+def binary_search(array: list, lower_bound: int, upper_bound: int, value: int) -> bool:
+    """
+    This function carries out Binary search on a 1d array
+    Array: A 1d sorted array
+    lower_bound: the lowest index for the binary search algorithm
+    upper_bound: the highest index for the binary search algorithm
+    value : the value ment to be searched
+    matrix = [1, 4, 7, 11, 15]
+    target = 1
+    >>> binary_search(matrix, 0, len(matrix) - 1, 1)
+    True
+    target = 23
+    >>> binary_search(matrix, 0, len(matrix) - 1, 1) 

```

Comment:
```suggestion
    >>> binary_search(matrix, 0, len(matrix) - 1, 1)
```
Remove the space at the end of the line or get psf/black to do it for you.  See CONTRIBUTING.md.